extensions/common/context.cc

```diff
@@ -294,6 +294,27 @@ void populateHTTPRequestInfo(bool outbound, bool use_host_header_fallback,
   }
 }
 
+void populateExtendedHTTPRequestInfo(RequestInfo* request_info) {
+  getValue({"source", "address"}, &request_info->source_address);
+  getValue({"destination", "address"}, &request_info->destination_address);
+
+  getValue({"request", "referer"}, &request_info->referer);
+  getValue({"request", "user_agent"}, &request_info->user_agent);
+  getValue({"request", "id"}, &request_info->request_id);
+  std::string trace_sampled;
+  getValue({"request", "headers", "x-b3-sampled"}, &trace_sampled);
```

Comment:
Does getValue return a bool or similar ?

Reply:
Yes, good point. Added it to the condition.